sockeye/train.py

```diff
@@ -497,20 +554,46 @@ def get_num_embed(args: argparse.Namespace) -> Tuple[int, int]:
                             "has to be even and a multiple of encoder attention heads (%d)" % (
                                 adjusted_transformer_encoder_model_size, args.transformer_attention_heads[0]))
 
+    if not num_embed_source:
+        num_embed_source = C.DEFAULT_NUM_EMBED
+
     if args.decoder == C.TRANSFORMER_TYPE:
         transformer_model_size_target = args.transformer_model_size[1]
+        total_target_factor_size = 0
+        for factor_combine, factor_size in zip(args.target_factors_combine, args.target_factors_num_embed):
+            if factor_combine == C.FACTORS_COMBINE_CONCAT:
+                total_target_factor_size += factor_size
+
         if not num_embed_target:
             logger.info("Target embedding size was not set it will automatically be adjusted to match the "
                         "Transformer target model size (%d).", transformer_model_size_target)
             num_embed_target = transformer_model_size_target
         else:
             # Make sure that if the user sets num_embed it matches the Transformer model size
-            check_condition(args.transformer_model_size[1] == num_embed_target,
+            check_condition(args.transformer_model_size[1] == num_embed_target + total_target_factor_size,
                             "Target embedding size must match transformer model size: %s vs. %s"
-                            % (args.transformer_model_size[1], num_embed_target))
+                            % (args.transformer_model_size[1], num_embed_target + total_target_factor_size))
+
+        total_target_factor_size = 0
```

Comment:
This block seems to be duplicate compared to lines 562 to 566, right? (maybe I'm missing something)

Reply:
good catch, must be a leftover from a previous merge. removed the second block.